File: fabric/common/src/main/java/com/noxcrew/noxesium/fabric/network/clientbound/ClientboundResetServerRulesPacket.java
package com.noxcrew.noxesium.fabric.network.clientbound;

import com.noxcrew.noxesium.api.fabric.network.NoxesiumPacket;
import com.noxcrew.noxesium.api.fabric.network.NoxesiumPayloadType;
import com.noxcrew.noxesium.fabric.network.CommonPackets;
import it.unimi.dsi.fastutil.ints.IntList;
import net.minecraft.network.RegistryFriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;

/**
 * Resets the stored value for one or more server rules.
 */
public record ClientboundResetServerRulesPacket(IntList indices) implements NoxesiumPacket {
    public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundResetServerRulesPacket> STREAM_CODEC =
            CustomPacketPayload.codec(ClientboundResetServerRulesPacket::write, ClientboundResetServerRulesPacket::new);

    private ClientboundResetServerRulesPacket(RegistryFriendlyByteBuf buf) {
        this(buf.readIntIdList());
    }

    private void write(RegistryFriendlyByteBuf buf) {
        buf.writeIntIdList(indices);
    }

    @Override
    public NoxesiumPayloadType<?> noxesiumType() {
        return CommonPackets.INSTANCE.CLIENT_RESET_SERVER_RULES;
    }
}

File: paper/legacy/src/main/kotlin/com/noxcrew/noxesium/legacy/paper/api/network/clientbound/ClientboundResetServerRulesPacket.kt
package com.noxcrew.noxesium.legacy.paper.api.network.clientbound

import com.noxcrew.noxesium.legacy.paper.api.network.NoxesiumPacket
import com.noxcrew.noxesium.legacy.paper.api.network.NoxesiumPackets
import it.unimi.dsi.fastutil.ints.IntList

/**
 * Resets the stored value for one or more server rules.
 */
public data class ClientboundResetServerRulesPacket(
    public val indices: IntList,
) : NoxesiumPacket(NoxesiumPackets.CLIENT_RESET_SERVER_RULES)
